opencog/atoms/value/FloatValue.cc

```diff
@@ -35,16 +36,10 @@ bool FloatValue::operator==(const Value& other) const
 	if (_value.size() != fov->_value.size()) return false;
 	size_t len = _value.size();
 	for (size_t i=0; i<len; i++)
-		// Compare floats with ULPS, because they are lexicographically
-		// ordered. For technical explanation, see
-		// http://www.cygnus-software.com/papers/comparingfloats/Comparing%20floating%20point%20numbers.htm
-		// if (1.0e-15 < fabs(1.0 - fov->_value[i]/_value[i])) return false;
-#define MAX_ULPS 24
-		if (MAX_ULPS < llabs(*(int64_t*) &(_value[i]) - *(int64_t*)&(fov->_value[i])))
+		if (not is_approx_eq_ulp(_value[i],fov->_value[i],24))
 			return false;
```

Comment:
this change should not be a part of this pull req; its completely unrelated.

Reply:
anyway, you should not remove the first half of the comment, and you should not hard-code the magic number 24.  The `MAX_ULPS` define was specifying that literal; its bad C/C++ style to have literals that don't have an associated `#define`  (because if some future programmer wants to change the 24, they have to find all other uses of 24 in the code base, and wonder if they have to change them too.  If they forget to change one of the other 24's, bugs will typically happen.)